core/shared/src/main/scala/org/http4s/multipart/MultipartReceiver.scala

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.multipart
+
+import cats.Applicative
+import cats.effect.Sync
+import cats.syntax.foldable.*
+import fs2.io.file.Files
+import org.http4s.DecodeFailure
+import org.http4s.Headers
+import org.http4s.InvalidMessageBodyFailure
+import org.http4s.headers.`Content-Disposition`
+import org.typelevel.ci.*
+
+trait MultipartReceiver[F[_], A] { self =>
+  type Partial
+  def decide(partHeaders: Headers): Option[PartReceiver[F, Partial]]
+  def assemble(partials: List[Partial]): Either[DecodeFailure, A]
```

Comment:
It'd be beneficial to add some scaladocs to clarify these methods and explain the meaning of Partial.